### ISSUE TITLE
Dynamically changed Handles cause a break

### ISSUE THREAD

**Issue by underGhost** [wbkd/react-flow#545]
Hey!
First off let me say thanks for creating this awesome package!

I have different `NodeTypes` that have different amount of `<Handles/>` on them. There is one component in particular that you make some selections, and based off the selections you make, it changes the elements `type`.  The re-rendering happens and everything looks as it should, but when you select a `Handle`, it errors out.

I've been able to trace it to:
1. The Node originally renders with `0` Handles.
2. Make sidebar selections.
3. Node changes type to where it now has `2` Handles.
4. Those handles can no longer be connected and error out.

The error I receive is:
```
ReactFlow.esm.js:7195 Uncaught TypeError: Cannot read property '0' of null
    at ConnectionLine (ReactFlow.esm.js:7195)
    at renderWithHooks (react-dom.development.js:14826)
    at updateFunctionComponent (react-dom.development.js:17059)
    at beginWork (react-dom.development.js:18644)
    at HTMLUnknownElement.callCallback (react-dom.development.js:189)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:238)
    at invokeGuardedCallback (react-dom.development.js:293)
    at beginWork$1 (react-dom.development.js:23235)
    at performUnitOfWork (react-dom.development.js:22186)
    at workLoopSync (react-dom.development.js:22162)
    at performSyncWorkOnRoot (react-dom.development.js:21788)
    at react-dom.development.js:11112
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11062)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11107)
    at flushSyncCallbackQueue (react-dom.development.js:11095)
    at flushPendingDiscreteUpdates (react-dom.development.js:21879)
    at flushDiscreteUpdates (react-dom.development.js:21859)
    at flushDiscreteUpdatesIfNeeded (react-dom.development.js:1095)
    at dispatchDiscreteEvent (react-dom.development.js:4168)```

**Comment by moklick** [comment]
Hey @underGhost 

This is an interesting case :) I will check it tomorrow.

**Comment by underGhost** [comment]
Thanks so much!

Yea I started to trace it down and it looks like, we `querySelectorAll` for the Handles only once. If it happened whenever the `type` changed then it would probably fix it.

**Comment by underGhost** [comment]
Was able to get it working by commenting out:
https://github.com/wbkd/react-flow/blob/541ea89cd7d7937de2f32c07e17ccb362724afdb/src/store/index.ts#L220-L226

Is it possible to have this updated? Or is there some work around I can implement?

**Comment by moklick** [comment]
We released a change in v6.0.3. Can you check if it works?

**Comment by underGhost** [comment]
Thanks @moklick! Looks like that fixed my issue. Appreciate you getting to it so fast!
